scripts/west_commands/runners/openocd.py

```diff
@@ -33,13 +33,14 @@ def __init__(self, cfg, pre_init=None, pre_load=None,
         if not config:
             default = path.join(cfg.board_dir, 'support', 'openocd.cfg')
             if path.exists(default):
-                config = default
+                config = [default]
         self.openocd_config = config
 
-        if self.openocd_config is not None and path.exists(self.openocd_config):
-            search_args = ['-s', path.dirname(self.openocd_config)]
-        else:
-            search_args = []
+        search_args = []
+        for i in self.openocd_config:
+            if path.exists(i):
+                search_args.append('-s')
+                search_args.append(path.dirname(i))
```

Comment:
I'm a bit nervous about what happens if two different directories added this way have the same file now. What happens if `-f` args are absolute; do we need `-s` in that case?

Reply:
This should be no different than having identically named files in multiple (implicit or explicit) search directories with the current runner. A list of the implicit search directories can be found here: http://www.openocd.org/doc/html/Running.html#Running

I would say that we do need `-s` for absolute `-f` arguments as the absolute file may have its own dependencies with relative paths.

Reply:
Another option could to stick with the current `-s` behavior and add yet another argument for explicitly adding search paths.

Reply:
I would also prefer if only `path.join(cfg.board_dir, 'support')` would be added to the search path automatically, if the openocd.cfg file is present. The other search paths should be given explicitly via a new parameter.

Reply:
I came across an issue trying to implement your suggestion:

The existing runner checks if the (single) configuration file path given is an existing file. It it is, it adds the directory containing that file to the search path (even if that file was not the default `<board>/support/openocd.cfg` file).

Now, my current approach to this was to extend this to all configuration files specified; the directory of any configurations file, which represents an existing file (meaning it is not a file just meant to found using the search directories), is added to the search path.

Limiting this to the original solution proved harder than I expected. What should the behaviour be when passing multiple files, where none of them is the default file? Adding only the first to the search path seems illogical; adding none of them breaks backwards compatibility. Adding only `<board>/support` breaks backwards compatibility.

For the sake of simplicity and backwards compatibility, I think my initial proposal of adding all to the search path is the best solution. I regard the risk of file name collisions to be low, and it is also present in the current implementation.

Ideas and comments are most welcome.

Reply:
@henrikbrixandersen that's a good argument; I'm convinced.



> I would also prefer if only `path.join(cfg.board_dir, 'support')` would be added to the search path automatically, if the openocd.cfg file is present.

@chrta that looks like what the code is already doing, so you are not requesting changes, right?

Reply:
@henrikbrixandersen I agree and  also think that the current approach you implemented is fine.